source/Octodiff/Core/SupportedAlgorithms.cs

```diff
@@ -21,13 +21,22 @@ public static IHashAlgorithm Create(string algorithm)
                 if (algorithm == "SHA1")
                     return Sha1();
 
-                throw new CompatibilityException(string.Format("The hash algorithm '{0}' is not supported in this version of Octodiff", algorithm));
+                throw new CompatibilityException(
+                    $"The hash algorithm '{algorithm}' is not supported in this version of Octodiff");
             }
         }
 
         public static class Checksum
         {
-            public static IRollingChecksum Adler32Rolling() { return new Adler32RollingChecksum();  }
+#pragma warning disable 618
+            public static IRollingChecksum Adler32Rolling(bool useV2 = false)
```

Comment:
This method seems redundant to me, why not just return the correct version in `Create`?

Reply:
The create method is only really used when reading an Octodiff file, it reads a string (name of checksum algorithm) in the header of the file that is then passed to Create to determine what checksum algorithm should be used, I left the Adler32Rolling() as is with the optional param for V2 as its public and I'm not sure who else is referencing Octodiff.